### PR TITLE
feat: Implement create news draft article - EXO-69636

### PR DESCRIPTION
**Pull request by sofyenne**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java [hunk 1]
```diff
@@ -16,31 +16,91 @@
...
 public class NewsServiceImplV2 implements NewsService {
 
-  public SpaceService spaceService;
+  public static final String       ARTICLES_ROOT_PAGE_NAME         = "Articles";
+
+  public static final MetadataType METADATA_TYPE                   = new MetadataType(1000, "news");
+
+  public static final String       NEWS_METADATA_DRAFT_OBJECT_TYPE = "newsDraftPage";
+
+  private static final Log         LOG                             = ExoLogger.getLogger(NewsServiceImpl.class);
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java]:
```suggestion
  private static final Log         LOG                             = ExoLogger.getLogger(NewsServiceImplV2.class);
```

File: services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java [hunk 2]
```diff
@@ -16,31 +16,91 @@
...
+import org.exoplatform.upload.UploadService;
+import org.exoplatform.wiki.WikiException;
+import org.exoplatform.wiki.model.DraftPage;
+import org.exoplatform.wiki.model.Page;
+import org.exoplatform.wiki.service.NoteService;
 
 public class NewsServiceImplV2 implements NewsService {
 
-  public SpaceService spaceService;
+  public static final String       ARTICLES_ROOT_PAGE_NAME         = "Articles";
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java]:
```suggestion
  public static final String       NEWS_ARTICLES_ROOT_NOTE_PAGE_NAME         = "Articles";
```

**Review by azayati — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java [hunk 3]
```diff
@@ -167,4 +228,91 @@ public void unarchiveNews(String newsId, String currentUserName) throws Exceptio
...
+        // save illustration
+        Long illustrationId = -1L;
+        if (news.getUploadId() != null && !news.getUploadId().isBlank()) {
+          illustrationId = saveNewsIllustration(news.getUploadId(), null);
+        }
+        Map<String, String> properties = new HashMap<>();
+        properties.put("summary", news.getSummary());
+        properties.put("publicationState", news.getPublicationState());
+        properties.put("uploadId", news.getUploadId());
+        properties.put("illustrationId", illustrationId >= 0 ? String.valueOf(illustrationId) : null);
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java]: - define constants for "summary", "illustrationId"
- "uploadId" and "publicationState" not needed IMO at this step

File: services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java [hunk 4]
```diff
@@ -167,4 +228,91 @@ public void unarchiveNews(String newsId, String currentUserName) throws Exceptio
...
+        NewsDraftObject newsDraftMetaDataObject =
+                                                new NewsDraftObject(NEWS_METADATA_DRAFT_OBJECT_TYPE, draftArticle.getId(), null);
+        MetadataKey newsDraftMetadataKey = new MetadataKey(METADATA_TYPE.getName(), "news", 0);
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java]:
```suggestion
        MetadataKey newsDraftMetadataKey = new MetadataKey(NEWS_METADATA_TYPE.getName(), "news", 0);
```
define a constant NEWS_METADATA_NAME for "news"

File: services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java [hunk 5]
```diff
@@ -167,4 +228,91 @@ public void unarchiveNews(String newsId, String currentUserName) throws Exceptio
...
+        if (illustrationId !=null) {
+          FileItem fileItem = fileService.getFile(illustrationId);
+          if (fileItem != null) {
+            news.setIllustration(fileItem.getAsByte());
+          }
+        }
+        return news;
+      }
+    }
+    throw new EntityNotFoundException("Parent note not found");
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java]:
```suggestion
    throw new EntityNotFoundException("News articles root note parent not found");
```

File: services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java [hunk 6]
```diff
@@ -167,4 +228,91 @@ public void unarchiveNews(String newsId, String currentUserName) throws Exceptio
...
+    UploadResource uploadResource = uploadService.getUploadResource(uploadId);
+    if (uploadResource == null) {
+      throw new IllegalStateException("Can't find uploaded resource with id : " + uploadId);
+    }
+    try {
+      InputStream inputStream = new FileInputStream(uploadResource.getStoreLocation());
+      FileItem fileItem = new FileItem(null,
+                                       uploadResource.getFileName(),
+                                       uploadResource.getMimeType(),
+                                       FILE_API_NAME_SPACE,
```
Comment (azayati) [inline services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java:307]:
```suggestion
                                       FILE_API_NAME_SPACE,
```
define a namespace for news

File: services/src/test/java/org/exoplatform/news/service/impl/NewsServiceImplV2Test.java [hunk 7]
```diff
@@ -0,0 +1,94 @@
+package org.exoplatform.news.service.impl;
```
Comment (azayati) [inline services/src/test/java/org/exoplatform/news/service/impl/NewsServiceImplV2Test.java:17]: add header

**Review by azayati — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java (hunk 3 above)
Comment (sofyenne) [reply services/src/main/java/org/exoplatform/news/service/impl/NewsServiceImplV2.java]: done!

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**